octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/DescriptionComposite.java

```diff
@@ -1,69 +1,160 @@
+import com.hpe.adm.nga.sdk.authentication.SimpleUserAuthentication;
+import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.octane.ideplugins.services.util.Util;
+import com.hpe.octane.ideplugins.eclipse.Activator;
+import com.hpe.octane.ideplugins.eclipse.ui.entitydetail.model.EntityModelWrapper;
+import com.hpe.octane.ideplugins.eclipse.ui.util.ClientLoginCookie;
+import com.hpe.octane.ideplugins.eclipse.ui.util.LinkInterceptListener;
+import com.hpe.octane.ideplugins.eclipse.ui.util.PropagateScrollBrowserFactory;
+import com.hpe.octane.ideplugins.eclipse.ui.util.StackLayoutComposite;
+import com.hpe.octane.ideplugins.eclipse.ui.util.resource.PlatformResourcesManager;
+import com.hpe.octane.ideplugins.eclipse.util.EntityFieldsConstants;
+
+public class DescriptionComposite extends Composite {
+
+    private PropagateScrollBrowserFactory factory = new PropagateScrollBrowserFactory();
+    private Color foregroundColor = PlatformResourcesManager.getPlatformForegroundColor();
+    private Color backgroundColor = PlatformResourcesManager.getPlatformBackgroundColor();
+    private Browser browserDescHtml;
+    private File octanePhotosDir;
+
+    public DescriptionComposite(Composite parent, int style) {
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
+        StackLayoutComposite stackLayoutComposite = new StackLayoutComposite(this, SWT.NONE);
+        browserDescHtml = factory.createBrowser(stackLayoutComposite, SWT.NONE);
+        browserDescHtml.addLocationListener(new LinkInterceptListener());
+        stackLayoutComposite.showControl(browserDescHtml);
+    }
+
+    public void setEntityModel(EntityModelWrapper entityModelWrapper) {
+        browserDescHtml.setText(getBrowserText(entityModelWrapper.getReadOnlyEntityModel()));
+    }
+
+    public String getBrowserText(EntityModel entityModel) {
+        String descriptionFromServerRemodeled = Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION)));
+        try {
+            descriptionFromServerRemodeled = downloadPictures(
+                    Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION))));
+        } catch (IOException e) {
+        }
+
+        String descriptionText = "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">"
+                + descriptionFromServerRemodeled
+                + "</font></body></html>";
+        if (descriptionText.equals("<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">" + "</font></body></html>")) {
+
+            return "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                    + getRgbString(foregroundColor) + ">" + "No description" + "</font></body></html>";
+        } else {
+            return descriptionText;
```

Comment:
@vladbenchea I know that you didn't write the code computing "descriptionText", but since you are touching it now we can improve it.
I'm not sure how often the above code gets called, but perhaps we can use StringBuilder instead of string.concatenation.

Instead of "if (descriptionText.equals("<html..." and constructing the result 2 (or 3) times, can't we just check for "if (descriptionFromServerRemodeled == "") and then construct the final description text?

Reply:
I will check this with Tibi as we will try to move part of this code in the common project. 
Probably it will be a better idea to have it organized with StringBuilder.